contribs/gmf/src/directives/layertree.js

```diff
@@ -175,6 +176,46 @@ gmf.LayertreeController = function($http, $sce, $scope, ngeoCreatePopup,
    */
   this.dataLayerGroup_ = this.layerHelper_.getGroupFromMap(this.map,
         gmf.DATALAYERGROUP_NAME);
+
+  $scope.$watchCollection('gmfLayertreeCtrl.tree.children', function(newNodes, oldNodes) {
+    var ctrl = $scope['gmfLayertreeCtrl'];
+    var reorderedLayers = ctrl.reorderLayer_(newNodes, oldNodes, ctrl.dataLayerGroup_.getLayers());
+    if (reorderedLayers) {
+      ctrl.dataLayerGroup_.setLayers(reorderedLayers);
+    }
+  });
+};
```

Comment:
I'm not a fan of using the controller's alias here. Instead, I'd rather use `this` since we're in the controller itself.

```javascript

  /**
   * @type {GmfThemesNode}
   */
  this.tree;

  $scope.$watchCollection(function() {
    if (this.tree) {
      return this.tree.children;
    }
  }.bind(this), function(newNodes, oldNodes) {
    var reorderedLayers = this.reorderLayer_(newNodes, oldNodes, this.dataLayerGroup_.getLayers());
    if (reorderedLayers) {
      this.dataLayerGroup_.setLayers(reorderedLayers);
    }
  }.bind(this));
```